pkg/config/manager.go

```diff
@@ -1,37 +1,181 @@
 package config
 
 import (
+	"context"
+	"errors"
+
 	"github.com/sirupsen/logrus"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/strings/slices"
 )
 
 // PersistenceManager manages persistence of the configuration.
 type PersistenceManager struct {
 	log    logrus.FieldLogger
+	cfg    PersistentConfig
 	k8sCli kubernetes.Interface
 }
 
+// ErrUnsupportedPlatform is an error returned when a platform is not supported.
+var ErrUnsupportedPlatform = errors.New("unsupported platform to persist data")
+
 // NewManager creates a new PersistenceManager instance.
-func NewManager(log logrus.FieldLogger, k8sCli kubernetes.Interface) *PersistenceManager {
-	return &PersistenceManager{log: log, k8sCli: k8sCli}
+func NewManager(log logrus.FieldLogger, cfg PersistentConfig, k8sCli kubernetes.Interface) *PersistenceManager {
+	return &PersistenceManager{
+		log:    log,
+		cfg:    cfg,
+		k8sCli: k8sCli,
+	}
 }
 
 // PersistSourceBindings persists source bindings configuration for a given channel in a given platform.
-func (m *PersistenceManager) PersistSourceBindings(commGroupName string, platform CommPlatformIntegration, channelName string, sourceBindings []string) error {
-	// TODO: Implement this as a part of https://github.com/kubeshop/botkube/issues/704
+func (m *PersistenceManager) PersistSourceBindings(ctx context.Context, commGroupName string, platform CommPlatformIntegration, channelAlias string, sourceBindings []string) error {
```

Comment:
Quick question: 

Other than the tests and the`ConfigPersistenceManager` interface, is there anywhere else in the code that uses this method?

Maybe I missed it.

Reply:
Very good question - I'm sorry I forgot to describe it: this will be used by @mszostok here: #747
(Not sure which PR comes first, but we agreed together on the interface). So it's actually a part of the "source bindings edit" feature 🙂

Reply:
Update: looks like Mateusz merged his PR, so after I rebase this PR this part will persist source bindings from the `edit` command 👍